From a91c07208f9a07ba75eab743a1152efa8e677f50 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 25 Jan 2018 00:58:09 +0000
Subject: [PATCH] interfaces/builtin: Replace Solus support with GLVND support

The legacy `update-alternatives` style path is now, and the new GLVND
system is used, which leverages indirection via `libGL,libGLX*` libraries.
This change allows the libGLX glvnd library to be read and thus GL using
snaps work again (tested with Steam, for 32-bit and 64-bit support).

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 interfaces/builtin/opengl.go | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/interfaces/builtin/opengl.go b/interfaces/builtin/opengl.go
index 93ccfa3e1..bd6588157 100644
--- a/interfaces/builtin/opengl.go
+++ b/interfaces/builtin/opengl.go
@@ -35,9 +35,6 @@ const openglConnectedPlugAppArmor = `
 /var/lib/snapd/lib/gl{,32}/ r,
 /var/lib/snapd/lib/gl{,32}/** rm,
 
-# Supports linux-driver-management from Solus (staged symlink trees into libdirs)
-/var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}glx-provider/**.so{,.*}  rm,
-
 # Bi-arch distribution nvidia support
 /var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libcuda*.so{,.*} rm,
 /var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libnvidia*.so{,.*} rm,
@@ -51,7 +48,7 @@ const openglConnectedPlugAppArmor = `
 /var/lib/snapd/hostfs/usr/share/vulkan/icd.d/*nvidia*.json r,
 
 # Main bi-arch GL libraries
-/var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}lib{GL,EGL}.so{,.*} rm,
+/var/lib/snapd/hostfs/{,usr/}lib{,32,64,x32}/{,@{multiarch}/}lib{GL,EGL,GLX}.so{,.*} rm,
 
 /dev/dri/ r,
 /dev/dri/card0 rw,
-- 
2.16.1

